### DISCUSSION TITLE
Best protocol? Is there such a thing?

### DISCUSSION THREAD

**Discussion by steveveach** [EdgeTX/edgetx#2468] [Q&A]
Hello everyone. I'm a newbie to EdgeTX and have come back to the hobby after a few years away and am blown away by the changes in technology. I own a RadioMaster TX16S Mark II Max with what they call their 4-in-1 internal module which of course, is the multi-protocol module.

Having such an amazing radio (thank you RadioMaster hardware engineers) with such an amazing operating system and selection of protocols (thank you EdgeTX open source developers) leaves me with a dilemma. **I can't choose a receiver and receiver protocol!** I've got too many choices. :)

Does anyone know if an analysis has ever been done comparing the protocols from a technical perspective--reliability, advantages, compatibility, etc.?

Thanks so much for any insight and helping me get out of my analysis paralysis. :)

**Comment by pfeerick** [comment]
I'm afraid EdgeTX can't take the credit re: protocols... that belongs to the [4in1 Multi-Protocol Module (MPM)]( https://github.com/pascallanger/DIY-Multiprotocol-TX-Module) project ;) 

As far as the best protocol - I think it really depends on who you ask. I'm sure I could talk all day about why ExpressLRS is the protocol for you... and then someone else will come along and say no... it's Futaba SFHSS... and then someone else will argue that it's Spektrum DSMX... Flysky AFHD2A/AFHD3, Frsky D16, etc, etc. 

At the end of the day... it doesn't really matter with the modern RF protocols... they all support telemetry - so you can get feedback as to if the receiver is going out of range, as well as any sensors you might install in your craft. However, you really need to start with "what do you want to fly... i.e. are we talking about planes with PWM outputs, or do we need SBUS, is there a flight controller involved. Or is it a helicopter (they are a beast I don't know much about, but I hear SBUS and the like are preferred there)... then there are multirotor... here ExpressLRS/ELRS is taking over. And with ELRSv3, it's also on the cusp of being ready for planes and PWM/servo control also. I've personally flown with Spektrum DSM2 (no telemetry), Frsky D8/D16 and Flysky AFHD2A, which are all supported by the MPM, and all work great. 

Of those three, I've come to prefer Flysky AFHD2A most (and am also playing with their newer ASHD3 as well)... just seems like they make good hardware, and have a decent range of sensors. Frsky was good and was my preference over Spektrum... which I started out with because of the cheap clone receivers, but is also a bit tricky as you have to make sure your MPM has been frequency tuned, and there is the hoops of what firmware is the receiver running v1/v2/LBT/FCC etc...  If you were to be interested in ExpressLRS, you'd need a separate RF module. I would suggest you do keep a close eye on that space, and see if you think it is for you. All my new builds are likely to use ELRS - I fly mostly multirotor, and some new plane builds will use ELRS gear also.

**Comment by steveveach** [comment]
Perfect answer pfeerick and all makes sense. After posting here, I also posted on the Official RadioMaster RC Group Facebook page and as you predicted, got lots of feedback. I now have a full understanding of my options. I will repost my conclusion post from Facebook here. Thanks again for responding.

CONCLUSION
After  doing some research on ELRS (Express Long Range System) I've reached  the following opinions for my specific case of: flying only planes,  never using flight controllers, owning the RadioMaster TX16S Mark II  radio, always flying between 200 and 600 feet away.  Most are relatively minor cons and offset against the largest pros,  which appear to be reliability because of the added transmission power  and speed/refresh rates, potentially making response times faster.
And  for a little level-setting on the reliability and refresh rates part. I  have been flying planes for decades and have never had a fail-safe  occurrence or at least not one that I noticed for more than a half  second. I have also never had issues with response times being too slow  for my eyes/brain/muscles to properly control a plane in flight. If  anything, it's my eyes/brain/muscles that are too slow, not the radios  and servos.

1. There are relatively few brands and versions of ELRS receivers on the  market that do NOT have flight controllers and instead have PWM servo  connectors.
2. ELRS supports at most 12 channels currently because of bandwidth constraints.
3. ELRS modules require power from the transmitter battery, draining the  battery quicker than the internal multi-protocol module.
4. ELRS antennas are probably more difficult to position in small "park  flyer" planes because of their physical shape. Multi-protocol receivers  have one or two simple wires that can be routed anywhere.
5. If someone flies or plans on flying BNF (bind and fly) foam models,  those manufacturers all use standard receivers with one of the common  protocols that are supported by the multi-protocol open source team. You  would have to remove the receiver and replace it with an ELRS receiver  with PWM outputs.
6. An ELRS module would need to be purchased as an added cost between $35 and $65.
7. So  I don't particularly see a reason to switch to ELRS for me, unless I  start flying FPV quadcopters and get a commercial license so I can fly a  mile away.

**Reply by pfeerick** [reply]
Just some comments on that 
3 - depends on the power level - i.e. I flew a lot with Happymodule ES24TX module, and that *does not* require external battery connection. Not does it use excessively more power than the MPM. It really depends on the output power you are flying with. And neither does another ~~redacted not yet released~~ high-power ELRS module I have, if it's not set to high output modes. 
4 - depends on the antenna - you don't *have* to use t-style antenna - there are standard wire antenna also, as well as the SMD antennas. 

But certainly, for BNF, and PWM, ELRS is certainly not the way to go. In about six months, the PWM limitation will no longer be valid, as 5CH PWM is already on the market, 10CH is possible with an adapter board, and I know of 7-8 channel PWM receivers in prototyping stage right now. But this is why I said keep an eye on this space as once that limitation is gone, you then need to look at the total system cost - i.e. $35-$65 for the module, then the cost of receivers vs alternatives such as say Flysky AFHDS2A compatible RXs. But the beauty of this hobby is there is no single right answer... just the one that is right for you :)

**Comment by steveveach** [comment]
Thanks for the additional info. Well said pfeerick.

When you say no additional power is required for the Happymodule, do you mean that the pins that connect it to the transmitter on the back of my RadioMaster TX16S Mark II will supply power to the ELRS Happymodule instead of tapping off the battery in the transmitter?

**Reply by pfeerick** [reply]
Correct - since it maxes out at 250mw, there is no provision for external power, nor is it needed. And the same is the case for higher power modules that do support external power, since you can limit the maximum output power (thus the power needed via the module bay). 

e.g. for the BetaFPV 1W ELRS module:
> There is an XT30 connector for connecting an external battery to power the module when you use 500mW or higher output power, because the radio won’t be able to provide enough current. Any 2S Lipo or Li-on packs would do. Do not use 3S or above or you could damage the module. (source: [oscar lang review](oscarliang.com/betafpv-elrs-micro-tx-module-1w-edition/))

And similarly, for a different system, Crossfire:
> If your Radio can not handle the current for this or you don’t want to drain your radio battery as fast you need to attach an external battery to the XT30 connector. (source: [Crossfire manual, p41](https://www.team-blacksheep.com/tbs-crossfire-manual.pdf))

In other words, whether you need an external battery/direct battery connection is dependent on the radio, the output power you want to use, and how long you want the battery to last per charge.

**Reply by steveveach** [reply]
Got it. Perfect explanation. Thanks for enlightening me. Yes, I'll keep an eye on ELRS even though I don't particularly need it now. Thanks again.

**Comment by newriderca** [comment]
> Hello everyone. I'm a newbie to EdgeTX and have come back to the hobby after a few years away and am blown away by the changes in technology. I own a RadioMaster TX16S Mark II Max with what they call their 4-in-1 internal module which of course, is the multi-protocol module.
> 
> Having such an amazing radio (thank you RadioMaster hardware engineers) with such an amazing operating system and selection of protocols (thank you EdgeTX open source developers) leaves me with a dilemma. **I can't choose a receiver and receiver protocol!** I've got too many choices. :)
> 
> Does anyone know if an analysis has ever been done comparing the protocols from a technical perspective--reliability, advantages, compatibility, etc.?
> 
> Thanks so much for any insight and helping me get out of my analysis paralysis. :)

If you have to use 4-1 go for Frsky!!! Other wise Elrs protocol if u have a compatible module.

**Comment by EhAye** [comment]
ELRS is the new hotness. LoRa tech lets it get some pretty incredible ranges at low power. Its open hardware too. Which means you can get really good and cheap modules, and also really bad and cheap modules depending on manufacturer.
There's Ghost which has similar features to ELRS and a bit more newbie friendly
Crossfire is still used and is still a solid system. 
Tracer is good in its niche area.
All of those require an external module though. 

Frsky is probably the best option of all the 4-in-1 protocols. And you can flash some frsky RX modules to ELRS later, if you decide to get an ELRS module

You made the correct choice on internal module though. I love having the multiprotocol internal, because i can connect to nearly anything. Even flies my old Hubsan x4! Also flies my moblite7 and a few other indoor whoops with random protocols. Amazing tool to have. 
Radiomaster's internal ELRS is only 250mw. So if you ever decide to do longrange stuff you'd still need another external module anyway. And you'd need to swap in the multiprotocol module each time you want to use that. Internal MPM was the correct choice ;D
